demo/react-hooks.html

```diff
@@ -0,0 +1,94 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+  <meta charset="UTF-8" />
+  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+  <title>Gridstack.js React integration example</title>
+  <link rel="stylesheet" href="demo.css" />
+  <script src="../dist/gridstack-h5.js"></script>
+
+  <!-- Scripts to use react inside html -->
+  <script src="https://unpkg.com/react@16/umd/react.production.min.js"></script>
+  <script src="https://unpkg.com/react-dom@16/umd/react-dom.production.min.js"></script>
+  <script src="https://unpkg.com/babel-standalone@6.15.0/babel.min.js"></script>
+</head>
+
+<body>
+  <div id="root"></div>
+</body>
+
+<script type="text/babel">
+  const { useState, useEffect, createRef, useRef } = React
+
+  const Item = ({ id }) => <div>I am item: {id}</div>
+
+  const ControlledStack = ({ items, addItem }) => {
+    const refs = useRef({})
+
+    if (Object.keys(refs.current).length !== items.length) {
+      items.forEach(({ id }) => {
+        refs.current[id] = refs.current[id] || createRef()
+      })
+    }
+
+    useEffect(() => {
+      const grid = GridStack.init()
```

Comment:
are you calling init() multiple times ? not the right thing. call once and save the var instead. Also removing all then making widgets doesn't seem right. the Vue example does only on what got added...

Reply:
Okay. Easy to call init() just once. 

Looking at the Vue example I see that there is a button that adds an additional widget. However in my case that wouldn't work, as the component receives a list of items from a server that I would like to render as widgets. When I update the server, the component is re-rendered with an updated list of items that I would like to make into widgets. That is why I am using `grid.makeWidget(someRef.current)`

Reply:
if you're reading a list from a server, I would call `load()` instead of makeWidget(). ha but wait the DOM elements are already all there ? then I would wrap the loop with a batchUpdate()/commit() so layout is only done once...

Reply:
Thanks - works. Also, I see that NOT specifying the row in the JSX that there is no bug regarding errant order changes (the bug that I mentioned earlier)

Reply:
@adumesny if you need to have live React components, using `load()` isn't a very good option.  I guess you could render React components into the widgets using [ReactDOM.createPortal](https://reactjs.org/docs/portals.html) after `load()` creates the elements, but then you would still need other code to hook up widgets added later.